### PR TITLE
Split contribution guide into smaller sections.

### LINKED ISSUE #822: Split contribution guide into smaller sections
```Also, at this point the contributing guide is getting very large.
I think it's time to split it into smaller sections. this can be a followup diff.```

_Originally posted by @omry in https://github.com/facebookresearch/hydra/pull/819#discussion_r459835481_

### PR DESCRIPTION
**Pull request by shagunsodhani**
fixes #822

## Motivation

Split contribution guide into smaller sections.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

NA

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

#822

### CODE REVIEW

File: website/docs/development/documentation.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+```
+$ yarn start
+```
+
+This command starts a local development server and opens up a browser window. Most changes are reflected live without having to restart the server.
+
+For more details, refer [here](https://github.com/facebookresearch/hydra/blob/master/website/README.md).
+
+## NEWS Entries
```
Comment (omry) [inline website/docs/development/documentation.md:27]: Put news entries before websites. More people need to add news entries than to edit the website.

File: website/docs/development/overview.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+sidebar_label: Overview
+---
+
+This guide assumes you have forked and checked-out the [repository](https://github.com/facebookresearch/hydra).
+It is recommended that you install Hydra in a virtual environment like [conda](https://docs.conda.io/en/latest/) or [virtualenv](https://virtualenv.pypa.io/en/latest/).
+
+## Environment setup
+Install [Miniconda](https://docs.conda.io/en/latest/miniconda.html) and create an empty Conda environment with:
+```
+conda create -n hydra38 python=3.8 -y
```
Comment (omry) [inline website/docs/development/overview.md:13]:
```suggestion
conda create -n hydra38 python=3.8 -qy
```

File: website/docs/development/overview.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+<div class="alert alert--info" role="alert">
+<strong>NOTE</strong>:
+The core Hydra framework supports Python 3.6 or newer. You may need to create additional environments for different Python versions if CI detect issues on a supported version of Python.
+</div>
+<br/>
```
Comment (omry) [inline website/docs/development/overview.md:20]: Docusaurus now have better support for [admonitions](https://v2.docusaurus.io/docs/markdown-features/#calloutsadmonitions).
Try the new syntax.

File: website/docs/development/testing.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+`nox` will run all the configured sessions. You can see the full list of nox sessions with `nox -l` and run specific sessions
+with `nox -s NAME` (you may need to quote the session name in some cases)
+
+## With pytest
+Run `pytest` at the repository root to run all the Hydra core tests.
+To run the tests of individual plugins, use `pytest plugins/NAME`.
+<div class="alert alert--info" role="alert">
+<strong>NOTE</strong>:
+Some plugins support fewer versions of Python than the Hydra core.
+</div>
```
Comment (omry) [inline website/docs/development/testing.md:17]: here [too](https://v2.docusaurus.io/docs/markdown-features/#calloutsadmonitions).

File: website/docs/development/overview.md (hunk 3 above)
Comment (shagunsodhani) [reply website/docs/development/overview.md:20]: This is sweet!

**Comment by shagunsodhani** [comment]
CI failing on Ax plugin. Doesnt fail locally. Re-running

**Comment by shagunsodhani** [comment]
It was a flaky test it seems.

**Pull request merged**